sbin/send_email.py

```diff
@@ -132,14 +133,12 @@ def construct_mail_client():
 
 
 def get_email_queue(db):
-    """Get an active queue of emails waiting to be sent."""
-    result = db.execute(
-        "SELECT emails.id, emails.user_id, users.user_email, emails.subject," +
-        " emails.body FROM emails INNER JOIN users ON" +
-        " emails.user_id = users.user_id WHERE" +
-        " emails.sent is NULL AND emails.error = ''" +
-        " ORDER BY id LIMIT 100;")
-
+    """Get an active queue of internal emails waiting to be sent."""
+    query = """SELECT id, user_id, email_address, subject, body FROM emails
+    WHERE email_address SIMILAR TO '%%@(%%.{}|{})' AND sent is NULL AND
```

Comment:
This query should be prepared differently, a user could enter a string that breaks the query if they add single quotes to `EMAIL_INTERNAL_FORMAT`, not sure if its possible to trigger a SQL injection issue since the strings are expanded inside a similar to clause but it might be possible so generally, we should avoid constructing queries via concatenation and use parameterized queries which offload the query construction to sqlalchemy in this case.

Reply:
Yeah I see the possibility of SQL injection here although it would be from sysadmin. Are there any good examples of parameterized queries in the code for sqlalchemy? I looked for docs on this but nothing was too helpful for getting this to work.